include/ozo/detail/begin_statement_builder.h

```diff
@@ -0,0 +1,69 @@
+#pragma once
+
+#include <boost/hana/filter.hpp>
+#include <boost/hana/fold.hpp>
+#include <boost/hana/plus.hpp>
+#include <boost/hana/string.hpp>
+#include <boost/hana/tuple.hpp>
+#include <ozo/core/none.h>
+#include <ozo/core/options.h>
+#include <ozo/query_builder.h>
+#include <ozo/transaction_options.h>
+#include <type_traits>
+
+namespace ozo::detail {
+
+namespace hana = boost::hana;
+
+struct begin_statement_builder {
+    using opt = transaction_options;
+
+    template <typename Level>
+    constexpr static auto to_string(decltype(opt::isolation_level), Level level) {
+        [[maybe_unused]] const auto prefix = BOOST_HANA_STRING(" ISOLATION LEVEL ");
+        if constexpr (decltype(level == isolation_level::serializable)::value) {
+            return prefix + BOOST_HANA_STRING("SERIALIZABLE");
+        } else if constexpr (decltype(level == isolation_level::repeatable_read)::value) {
+            return prefix + BOOST_HANA_STRING("REPEATABLE READ");
+        } else if constexpr (decltype(level == isolation_level::read_committed)::value) {
+            return prefix + BOOST_HANA_STRING("READ COMMITTED");
+        } else if constexpr (decltype(level == isolation_level::read_uncommitted)::value) {
+            return prefix + BOOST_HANA_STRING("READ UNCOMMITTED");
+        } else {
```

Comment:
It is not really mandatory, but since `isolation_level` members are `hana::type` you can try to use [hana::map](https://www.boost.org/doc/libs/1_67_0/libs/hana/doc/html/structboost_1_1hana_1_1map.html) here for dispatching. I think you could like the results. :)